WFLY-21013 eliminate testAddingNewResources non determinism

18chanp1 opened the pull request:
https://issues.redhat.com/browse/WFLY-21013



YamlConfigurationExtensionTest is indeterministic because there is no guarantee of the order of operations after processOperations is run. 

The fix is to sort the list after running processOperations and then fixing the order instead.

yersan commented:
Thanks for contributing to WildFly Core @18chanp1 !

This Pull Request (PR) and #6534 are targeting the same issue and the same testcase, so it is preferable a single PR  instead of two.

Could you add all your changes in a single PR instead? 

We do not need two commits, but that's up to you. A single PR with a single commit is even better here since we can later easily backport it to other branches when required.

Also, notice that the commit message for each commit should [also reference a JIRA number](https://docs.wildfly.org/27/Hacking_On_WildFly.html#commit-message).

18chanp1 commented:
Hi @yersan, thanks for the suggestions

I've squashed the commits into 1 commit, combined it into this PR, added gaps. Please let me know if there are any additional changes to be made.

Thanks
